ssl_cert_tracker/db_helper.py

```diff
@@ -4,53 +4,81 @@
 import time
 import logging
 
-from django.db.models.signals import post_save
-
-from django.core.signals import request_finished
-from django.dispatch import receiver
-
 from django.core.exceptions import ObjectDoesNotExist
-from ssl_cert_tracker.models import TestCertsData, TestCertsHistory
+from ssl_cert_tracker.models import NmapCertsData, NmapHistory
+from django.conf import settings
 
-logging.basicConfig(filename='pocProject.log',level=logging.DEBUG)
+logging.basicConfig(filename='p_soc_auto.log', level=logging.DEBUG)
 
-def Insert_Into_CertsData(json_data):
-    """Insert_Into_CertsData """
-    return_code = "Success"
+def Insert_Into_CertsData(json_data): 
+    """Insert_Into_CertsData check if record already exist then update otherwise insert """
     try:
-        db_certs = TestCertsData()
+        cert_user = NmapCertsData.get_user()
+        cert_status = NmapCertsData.get_cert_state(json_data["orion_id"], json_data["md5"])
+        if cert_status not in [0, 1, 2]:
+            msg = "failure"
+            logging.error("Error accessing django model NmapCertsData get_cert_state:%s", msg)
+            return
+        elif cert_status == 0: #  cert has not changed
+            msg = "To be decided!"
+            logging.info("Cert data update:.... %s", msg)
+            return
+        elif cert_status == 1:# new reord
+            db_certs = NmapCertsData()
+            msg = "Unchanged!"
+            logging.info("Cert data info:.... %s", msg)
+        else: # cert changed cert_status == 2
+            db_certs = NmapCertsData.get_cert_handle(json_data["orion_id"])
+            msg = "Changed!"
+            logging.info("Cert data info:.... %s", msg)
+
         db_certs.orion_id = json_data["orion_id"]
         db_certs.addresses = json_data["addresses"]
         
-        if "not_before" in json_data and validate(json_data["not_before"]) == True:
-            not_after_posix = time.mktime(datetime.datetime.strptime(json_data["not_before"], "%Y-%m-%d").timetuple())
+        db_certs.created_by_id= cert_user.id
+        db_certs.updated_by_id= cert_user.id
+        
+        db_certs.updated_on = datetime.now().strftime ("%Y-%m-%d %H:%M:%S.%f")

```

Comment:
in django applications, please use the datetime classes and functions from django.utils.timezone instead of datetime